huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go

```diff
@@ -1031,12 +1039,88 @@ func resizeDcsInstance(ctx context.Context, d *schema.ResourceData, meta interfa
 		}
 		if instance.SpecCode != d.Get("flavor").(string) {
 			return fmt.Errorf("change flavor failed, after changed the DCS flavor still is: %s, expected: %s",
-				instance.SpecCode, specCode)
+				instance.SpecCode, newSpecCode)
 		}
 	}
 	return nil
 }
 
+func buildResizeInstanceOpt(client *golangsdk.ServiceClient, d *schema.ResourceData, oldSpecCode,
+	newSpecCode string) (*instances.ResizeInstanceOpts, error) {
+	opts := instances.ResizeInstanceOpts{
+		SpecCode:    newSpecCode,
+		NewCapacity: d.Get("capacity").(float64),
+	}
+	if d.Get("charging_mode").(string) == chargeModePrePaid {
+		opts.BssParam = instances.DcsBssParamOpts{
+			IsAutoPay: "true",
+		}
+	}
+	if oldSpecCode != newSpecCode {
+		oldFlavor, err := getFlavorBySpecCode(client, oldSpecCode)
+		if err != nil {
+			return nil, err
+		}
+		newFlavor, err := getFlavorBySpecCode(client, newSpecCode)
+		if err != nil {
+			return nil, err
+		}
+		changeType, err := getFlavorChangeType(oldFlavor, newFlavor)
+		if err != nil {
+			return nil, err
+		}
+		opts.ChangeType = changeType
+		if changeType == "createReplication" {
+			azCodes, err := getAzCode(d, client)
+			if err != nil {
+				return nil, err
+			}
+			opts.AvailableZones = azCodes
+		}
+		if changeType == "deleteReplication" {
+			if newFlavor.CacheMode == "ha" {
+				opts.NodeList = utils.ExpandToStringList(d.Get("deleted_nodes").([]interface{}))
+			} else {
```

Comment:
better to use `if` here, as there are several cache modes

Reply:
fixed